src/canvas/StaticCanvas.ts

```diff
@@ -885,10 +885,15 @@ export class StaticCanvas<
     }
     if (object) {
       ctx.save();
+      const { skipOffscreen } = this;
+      // if the object doesn't move with the viewport,
+      // the offscreen concept does not apply;
+      this.skipOffscreen = needsVpt;
```

Comment:
I am so confused.
Shouldn't it be the other way around? If it respects vpt it should be subject to offscreen optimizations.
Anyways I think tests will not pick this up because the bg/overlay objects have no canvas ref (which is dangerous as well), what makes the offscreen test skip entirely.

Reply:
Eh because skipOffscreen means skip rendering of offscreen object, not skip the check.
So if needsVpt, skip the rendering when offscreen.
For a moment i thought i got confused.

I can disable the line and see if tests fail